### DISCUSSION TITLE
2021-09-28 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#684] [Maintainer discussions]
### Discussion topics

- **[Updates](https://github.com/centerofci/mathesar/discussions/684#discussioncomment-1397028)**: General updates everyone should be aware of.
- **[Questions](https://github.com/centerofci/mathesar/discussions/684#discussioncomment-1397046)**: Questions for the team to answer.
- **[Active Work](https://github.com/centerofci/mathesar/discussions/684#discussioncomment-1397088)**: Work planned for the next week.
- **[Retrospective](https://github.com/centerofci/mathesar/discussions/684#discussioncomment-1397089)**: Summary of last week's work.

### About
Each Tuesday, we start a discussion about Mathesar's progress. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

**Comment by kgodey** [comment]
## Updates

### Roadmap

I've cut the following milestones from our alpha release plan in order to focus on what we can do well by December.

- [Computed Data](https://github.com/centerofci/mathesar/milestone/12)
- [Summarized Data](https://github.com/centerofci/mathesar/milestone/16)
- [Undo and Redo (DML)](https://github.com/centerofci/mathesar/milestone/25)
- 2021-11 Stability release (we'll work these issues into other milestones)

We discussed cutting these features during our State of Mathesar meeting but decided to keep them in and see how it went. Based on the last couple of weeks, it seems clear that we were being too optimistic :).

I've also updated the milestone deadlines on our [roadmap](https://github.com/centerofci/mathesar/milestones?direction=asc&sort=due_date&state=open). Please review and comment below about what you think.

### Team

Our hiring process for the frontend engineer has been completed. Our new hire starts on Oct 12, I'll do a proper introduction then.

We've also begun interviews for our [backend engineer](https://wiki.mathesar.org/jobs) position. We're still accepting applications and referrals are appreciated.

### Project Planning

We're now in [the new GitHub Issues](https://github.blog/2021-06-23-introducing-new-github-issues/) beta so I'm going to be moving Mathesar to the new project structure from our current [Active](https://github.com/centerofci/mathesar/projects/1), [Backlog](https://github.com/centerofci/mathesar/projects/2), and [Future](https://github.com/centerofci/mathesar/projects/3) boards.

I've started setting up a [Mathesar project](https://github.com/orgs/centerofci/projects/1) but I'll need to set up some automations and views before it's usable. It's not a high priority so I anticipate it might take a couple of weeks to move over.

**Reply by mathemancer** [reply]
> * [Computed Data](https://github.com/centerofci/mathesar/milestone/12)
> * [Summarized Data](https://github.com/centerofci/mathesar/milestone/16)
> * [Undo and Redo (DML)](https://github.com/centerofci/mathesar/milestone/25)

I think these are the right sacrifices to make at the moment.

**Comment by kgodey** [comment]
## Questions

@mathemancer, @powellc, @ghislaineguerin, @pavish: Your responses are encouraged. Others are also welcome to respond.

### Retrospective
- What went well over the past week?
- What could have gone better over the past week?
- Are there any improvements you'd like to suggest?

### Active Work
- Does your workload for this week look okay?
- We have two milestones due in the next two weeks. Are there any blockers to meeting these deadlines?

### Discussion
Please respond to the [Updates](https://github.com/centerofci/mathesar/discussions/684#discussioncomment-1397028) post directly if you'd like to discuss anything there. If you have any other topics to discuss, please post them as new replies to the [parent post](https://github.com/centerofci/mathesar/discussions/684#discussion-3596965).

**Reply by kgodey** [reply]
> What went well over the past week?
> What could have gone better over the past week?
> Are there any improvements you'd like to suggest?

N/A, I was taking time off.

> Does your workload for this week look okay?

It looks doable but I have interviews etc. this week, so I may not be able to get it all done.

> We have two milestones due in the next two weeks. Are there any blockers to meeting these deadlines?

We need to resolve the add records UX issues, but that's on my list so I don't think it's a blocker.

**Reply by mathemancer** [reply]
> What went well over the past week?

I clarified some things about the issues around server default setting, and found a promising direction to proceed on that.  The text types PR is submitted.

> What could have gone better over the past week?

Unfortunately, the server default setting problems bogged down the type implementation work.

> Are there any improvements you'd like to suggest?

Nothing specific.


> Does your workload for this week look okay?

Yes, I think so.  There may be some types pushed into the next week, especially the custom money type.

> We have two milestones due in the next two weeks. Are there any blockers to meeting these deadlines?

None that I can see.

**Reply by pavish** [reply]
> What went well over the past week?
> What could have gone better over the past week?

I think things went generally well during the last week. I was off for a couple days. There are no major updates.

> Are there any improvements you'd like to suggest?

None, at the moment

> Does your workload for this week look okay?

It looks okay to me.

> We have two milestones due in the next two weeks. Are there any blockers to meeting these deadlines?

None.

**Reply by ghislaineguerin** [reply]
> What went well over the past week?

It was good to have time to work on the UI and think about the UX in more detail. 

> What could have gone better over the past week?

I didn't make much progress on review rounds for views

> Are there any improvements you'd like to suggest?

I needed to switch gears from views to UI so I think looking at views w fresh eyes will help

> Does your workload for this week look okay?

It does

> We have two milestones due in the next two weeks. Are there any blockers to meeting these deadlines?

None

**Comment by kgodey** [comment]
## Active Work

This post lists work planned for this week (see also: [Active](https://github.com/centerofci/mathesar/projects/1)). It also includes a current summary of milestones involved in this week's work.

### [2021-10-01] [[05] Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **[9 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B05%5D%20Editable%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish] #197

### [2021-10-08] [[06] 2021-10 Stability](https://github.com/centerofci/mathesar/milestone/34)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B06%5D%202021-10%20Stability%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin] #473
  - [@pavish] #651
  - [@pavish] #639
- **Issues not yet started:**
  - [@kgodey] #592
  - [@kgodey] #280
  - [@kgodey] #583
  - [@ghislaineguerin] #636
  - [@ghislaineguerin] #638
  - [@pavish] #640
  - [@pavish] #641
  - [@pavish] #642
  - [@kgodey] #225
  - [@pavish] #643
  - [@pavish] #213
  - [@bohemia420] #635
  - [@bohemia420] #668
  - [@pavish] #619

### [2021-10-15] [[07] Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **[22 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B07%5D%20Working%20with%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@mr-gabe49] #448
  - [@dmos62] #498
  - [@mr-gabe49] #496
- **Issues remaining in backlog:**
  - 10 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%22%5B07%5D%20Working%20with%20Tables%22) issues

### [2021-10-22] [[08] Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **[30 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B08%5D%20Initial%20Data%20Types%22) already closed.** 
- **Issues currently in progress:**
  - [@mathemancer] #417
  - [@eito-fis] #426
  - [@powellc] #385
  - [@powellc] #387
  - [@mathemancer] #407
  - [@mathemancer] #405
  - [@mathemancer] #404
- **Issues not yet started:**
  - [@powellc] #384
  - [@mathemancer] #412
  - [@mathemancer] #424
  - [@mathemancer] #430
  - [@mathemancer] #420
  - [@powellc] #410
  - [@mathemancer] #681
  - [@mathemancer] #682
  - [@kgodey] #661
- **Issues remaining in backlog:**
  - 17 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%22%5B08%5D%20Initial%20Data%20Types%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%22%5B08%5D%20Initial%20Data%20Types%22) issues
  - 9 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%22%5B08%5D%20Initial%20Data%20Types%22) issues
  - 8 [product](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20product%22+milestone%3A%22%5B08%5D%20Initial%20Data%20Types%22) issues

### [2021-10-31] [[08A] 2021-10 improvements](https://github.com/centerofci/mathesar/milestone/31)

- **Issues currently in progress:**
  - [@arvind-1] #555
- **Issues not yet started:**
  - [@silentninja] #421
- **Issues remaining in backlog:**
  - 5 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%22%5B08A%5D%202021-10%20improvements%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%22%5B08A%5D%202021-10%20improvements%22) issues
  - 12 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%22%5B08A%5D%202021-10%20improvements%22) issues
  - 3 [infrastructure](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20infrastructure%22+milestone%3A%22%5B08A%5D%202021-10%20improvements%22) issues

### [2021-11-05] [[09] Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B09%5D%20Working%20with%20Views%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin] #455
  - [@ghislaineguerin] #443
  - [@ghislaineguerin] #466
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #243
  - [@ghislaineguerin] #456
- **Issues remaining in backlog:**
  - 7 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%22%5B09%5D%20Working%20with%20Views%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%22%5B09%5D%20Working%20with%20Views%22) issues
  - 2 [product](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20product%22+milestone%3A%22%5B09%5D%20Working%20with%20Views%22) issues

### [2021-11-19] [[11] Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **[7 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B11%5D%20Multiple%20Database%20Support%22) already closed.** 
- **Issues currently in progress:**
  - [@aaditkamat] #395
- **Issues remaining in backlog:**
  - 3 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%22%5B11%5D%20Multiple%20Database%20Support%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%22%5B11%5D%20Multiple%20Database%20Support%22) issues
  - 3 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%22%5B11%5D%20Multiple%20Database%20Support%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%22%5B11%5D%20Multiple%20Database%20Support%22) issues

**Comment by kgodey** [comment]
## Retrospective

This post lists issue updates from 2021-09-22 until now by milestone.

### [[05] Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **Issues started, not yet finished:**
  - [@pavish] #197

### [[06] 2021-10 Stability](https://github.com/centerofci/mathesar/milestone/34)

- **Issues closed:**
  - [@mr-gabe49] #646
- **Issues started, not yet finished:**
  - [@ghislaineguerin] #473
  - [@pavish] #651
  - [@pavish] #639

### [[07] Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **Issues closed:**
  - [@mathemancer] #663
  - [@pegnott] #495
- **Issues started, not yet finished:**
  - [@mr-gabe49] #448
  - [@dmos62] #498
  - [@mr-gabe49] #496

### [[08] Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **Issues started, not yet finished:**
  - [@mathemancer] #417
  - [@eito-fis] #426
  - [@powellc] #385
  - [@powellc] #387
  - [@mathemancer] #407
  - [@mathemancer] #405
  - [@mathemancer] #404
- **New issues created:**
  - #682, created by @mathemancer
  - #681, created by @mathemancer

### [[08A] 2021-10 improvements](https://github.com/centerofci/mathesar/milestone/31)

- **Issues started, not yet finished:**
  - [@arvind-1] #555
- **New issues created:**
  - #679, created by @bohemia420

### [[09] Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **Issues started, not yet finished:**
  - [@ghislaineguerin] #455
  - [@ghislaineguerin] #443
  - [@ghislaineguerin] #466
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #243
  - [@ghislaineguerin] #456

### [[11] Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **Issues started, not yet finished:**
  - [@aaditkamat] #395

### [[13A] 2021-11 improvements](https://github.com/centerofci/mathesar/milestone/32)

- **New issues created:**
  - #683, created by @pavish
